frontend/public/src/lib/util.js

```diff
@@ -283,3 +284,74 @@ export const intCheckFeatureFlag = (
 export const checkFeatureFlag = (flag: string, uniqueID: string | number) => {
   return intCheckFeatureFlag(flag, uniqueID, document, SETTINGS)
 }
+
+/**
+ * This is a comparison method used to sort an array of Course Runs
+ * from earliest start date to latest start date.
+ * @param {BaseCourseRun} courseRunA The first Course Run to compare.
+ * @param {BaseCourseRun} courseRunB The second Course Run to compare.
+ */
+export const compareCourseRunStartDates = (
+  courseRunA: BaseCourseRun,
+  courseRunB: BaseCourseRun
+) => {
+  if (moment(courseRunA.start_date).isBefore(courseRunB.start_date)) {
+    return -1
+  }
+  if (moment(courseRunA.start_date).isAfter(courseRunB.start_date)) {
+    return 1
+  }
+  // CourseRunA and CourseRunB share the same start date.
+  return 0
+}
+
+/**
+ * Returns the text to be displayed on a course catalog card's tag.
+ * This text will either be "Start Anytime" or "Start Date: <most recent, future, start date for the course>".
+ * If the Course has at least one associated Course Run which is not self-paced, and
+ * Course Run start date is in the future, then return "Start Date: <most recent, future, start date for the course>".
+ * If the Course has at least one associated Course Run which is not self-paced, and
+ * Course Run start date is in the past, then return "Start Anytime".
+ * If the course only has Course Runs which are self-paced, display "Start Anytime".
+ * If the isArchived flag is set, display "Course content available anytime".
+ * @param {CourseDetailWithRuns} course The course being evaluated.
+ * @param {isArchived} boolean Whether the course is archived or not.
+ * @param {showPast} boolean Render start dates for course start dates that are in the past.
```

Comment:
I think, if I follow correctly, that this should really return the closest to now course run.. generally this will only ever happen with a specific course run (this is a path that should only be taken when rendering things for the more dates code) but I added a reverse sort to grab the closest-to-now courserun if there aren't future ones.

Reply:
Makes sense.  Can you just update the param comment to indicate exactly what showPast does?